src/main/java/software/amazon/event/kafkaconnector/mapping/JsonPathDetailTypeMapper.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.event.kafkaconnector.mapping;
+
+import static com.jayway.jsonpath.Configuration.defaultConfiguration;
+import static com.jayway.jsonpath.Option.SUPPRESS_EXCEPTIONS;
+import static java.lang.String.format;
+import static java.util.Collections.singletonMap;
+import static org.apache.kafka.connect.json.JsonConverterConfig.SCHEMAS_ENABLE_CONFIG;
+import static software.amazon.event.kafkaconnector.EventBridgeSinkConfig.AWS_DETAIL_TYPES_MAPPER_JSON_PATH_MAPPER_FIELDREF;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.JsonPath;
+import org.apache.kafka.connect.json.JsonConverter;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.slf4j.Logger;
+import software.amazon.event.kafkaconnector.EventBridgeSinkConfig;
+import software.amazon.event.kafkaconnector.logging.ContextAwareLoggerFactory;
+
+public class JsonPathDetailTypeMapper implements DetailTypeMapper {
+  private static final Logger log =
+      ContextAwareLoggerFactory.getLogger(JsonPathDetailTypeMapper.class);
+
+  private static final Configuration jsonPathConfiguration =
+      defaultConfiguration()
+          .addOptions(
+              SUPPRESS_EXCEPTIONS // suppress exception otherwise
+              // com.jayway.jsonpath.ReadContext#read throws an exception if JSON path could
+              // not
+              // be found
+              );
+
+  private String jsonPath;
+  private final JsonConverter jsonConverter;
+  private final ObjectMapper objectMapper;
+
+  public JsonPathDetailTypeMapper() {
+    this.jsonConverter = new JsonConverter();
+    this.objectMapper = new ObjectMapper();
+    this.jsonConverter.configure(singletonMap(SCHEMAS_ENABLE_CONFIG, "false"), false);
+  }
+
+  @Override
+  public String getDetailType(SinkRecord record) {
+    if (record == null) {
+      throw new IllegalArgumentException("SinkRecord is null. Unable to extract detail type.");
+    }
+
+    if (record.topic() == null || record.topic().trim().isEmpty()) {
+      throw new IllegalArgumentException(
+          "SinkRecord topic is null or empty but is required for fallback logic.");
+    }
+
+    var topic = record.topic();
+    try {
+      var jsonBytes = jsonConverter.fromConnectData(topic, record.valueSchema(), record.value());
+      if (jsonBytes == null) {
+        // TODO: discuss throw or error
+        log.error(
+            "Record value conversion to JSON bytes returned null for record '{}', using topic '{}' as fallback",
+            record,
+            topic);
+        return topic;
+      }
+
+      var jsonNode = objectMapper.readTree(jsonBytes);
```

Comment:
Is jsonNode only needed for verification purposes?

Reply:
@embano1 did forget to delete it, it's verified by
```
    JsonPath.using(jsonPathConfiguration).parse(new String(jsonBytes))
```
some lines below

Reply:
Uff, yes removed!